internal/template/template.go

```diff
@@ -181,8 +181,35 @@ func (mock *{{$mock.MockName}}
 	mock.lock{{.Name}}.RUnlock()
 	return calls
 }
+{{- if $.WithResets}}
+// {{.Name}}ResetCalls Reset all the calls that were made to {{.Name}}.
+func (mock *{{$mock.MockName}}) {{.Name}}ResetCalls() {
+	mock.lock{{.Name}}.Lock()
+	mock.calls.{{.Name}} = []struct {
+		{{- range .Params}}
+		{{.Name | Exported}} {{.TypeString}}
+		{{- end}}
+	}{}
+	mock.lock{{.Name}}.Unlock()
+}
+{{end}}
+{{end -}}
+{{- if $.WithResets}}
+// ResetCalls reset all the calls that were made to all mocked methods.
+func (mock *{{$mock.MockName}}) ResetCalls() {
+	{{- range .Methods}}
+	mock.lock{{.Name}}.Lock()
+	mock.calls.{{.Name}} = []struct {
```

Comment:
This code generation could end up being quite a large number of lines. Would setting it to `nil` work just as well?

Reply:
I agree, I would just reset this to `nil`, `append is save on a `nil` slice.